Mage/src/main/java/mage/game/permanent/token/TokenImpl.java

```diff
@@ -375,6 +391,15 @@ private static void putOntoBattlefieldHelper(CreateTokenEvent event, Game game,
                 }
                 // end of aura code : just remove this line if everything works out well
 
+                if (permanentAttachedTo != null) {
+                    if (permanent.hasSubtype(SubType.AURA, game)) {
+                        permanent.getAbilities().get(0).getTargets().get(0).add(permanentAttachedTo.getId(), game);
+                        permanent.getAbilities().get(0).getEffects().get(0).apply(game, permanent.getAbilities().get(0));
+                    } else {
+                        permanentAttachedTo.addAttachment(permanent.getId(), source, game);
```

Comment:
Original `MaskToken` has `AttachEffect` and it works with players and cards too. Current code with `putOntoBattlefield` and `attachTo` param works with permanents only. It's not a critical for `MaskToken` but can broken for another tokens/cards after moving to new code (from apply effect to direct attach).

Another diff code from old effect -- it has zcc code checking (use cases: blink after activate, activate while on stack). But I don't know is it needs in new code.

Reply:
yeah I knew that going in and figured it was a bridge we could cross when we got to it since we don't know what kind of aura tokens we could potentially see in the future